packages/ripple-binary-codec/src/enums/index.ts

```diff
@@ -1,164 +1,30 @@
 import * as enums from './definitions.json'
-import { SerializedType } from '../types/serialized-type'
-import { Buffer } from 'buffer/'
-import { BytesList } from '../binary'
-
-/*
- * @brief: All valid transaction types
+import { RippledDefinitions, FieldInstance, Bytes } from './rippled-definitions'
+/**
+ * By default, coreTypes from the `types` folder is where known type definitions are initialized to avoid import cycles.
  */
-export const TRANSACTION_TYPES = Object.entries(enums.TRANSACTION_TYPES)
-  .filter(([_key, value]) => value >= 0)
-  .map(([key, _value]) => key)
+const DEFAULT_DEFINITIONS = new RippledDefinitions(enums, {})
```

Comment:
Why doesn't `DEFAULT_DEFINITIONS` have types?

Reply:
Circular dependency problem again - can't import `../types` in the same file that `DEFAULT_DEFINITIONS` is initialized since `DEFAULT_DEFINITIONS` is needed to form some of the types.

Reply:
Do we _need_ the types in XrplDefinitions? IMO it'll be pretty rare that someone needs to create a new type.
Perhaps instead of needing to add all the types, the `types` field accepts only extra types.